common/src/main/java/com/microsoft/identity/common/internal/cache/HelloCache.java

```diff
@@ -70,7 +80,29 @@ public static void setIsEnabled(final boolean value) {
     }
 
     /**
-     * Default constructor.
+     * Holds result of querying cache result.
+     */
+    @RequiredArgsConstructor
+    public static class HelloCacheResult {
```

Comment:
create a class with private constructor, but expose 
1. static HelloCacheResult createWithError()
2. static HelloCacheResult createWithProtocolVersion()
3. String getProtocolVersionOrThrow()

Alternatively... use Kotlin's Result<> class. not sure if it can be used in java. (should be)

Reply:
Java does not work well with kotlin's Result object; can't reference methods on kotlin's result class.

Reply:
Made suggested changes except that we don't throw from Hello* classes and only send error in result. Caller can decide based on result contents. BrokerMsalController creates UnsupportedBrokerException and returns.

Also moved entire code to kotlin.